packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -235,66 +288,100 @@ export class SummarizerNode implements IRootSummarizerNode {
+			});
+		}
 
-		if (proposalHandle !== undefined) {
-			const maybeSummaryNode = this.pendingSummaries.get(proposalHandle);
+		let result: RefreshSummaryResult | undefined;
+		try {
+			this.logger.sendTelemetryEvent({
+				eventName: "refreshLatestSummary_start",
+				proposalHandle,
+				referenceSequenceNumber: this.referenceSequenceNumber,
+				summaryRefSeq,
+			});
+
+			if (proposalHandle !== undefined) {
+				const maybeSummaryNode = this.pendingSummaries.get(proposalHandle);
+
+				if (maybeSummaryNode !== undefined) {
+					this.refreshLatestSummaryFromPending(
+						proposalHandle,
+						maybeSummaryNode.referenceSequenceNumber,
+					);
+					result = { latestSummaryUpdated: true, wasSummaryTracked: true, summaryRefSeq };
+					return result;
+				}
 
-			if (maybeSummaryNode !== undefined) {
-				this.refreshLatestSummaryFromPending(
+				const props = {
+					summaryRefSeq,
+					pendingSize: this.pendingSummaries.size ?? undefined,
+				};
+				this.logger.sendTelemetryEvent({
+					eventName: "PendingSummaryNotFound",
 					proposalHandle,
-					maybeSummaryNode.referenceSequenceNumber,
-				);
-				return { latestSummaryUpdated: true, wasSummaryTracked: true, summaryRefSeq };
+					referenceSequenceNumber: this.referenceSequenceNumber,
+					details: JSON.stringify(props),
+				});
 			}
 
-			const props = {
-				summaryRefSeq,
-				pendingSize: this.pendingSummaries.size ?? undefined,
+			// If the summary for which refresh is called is older than the latest tracked summary, ignore it.
+			if (this.referenceSequenceNumber >= summaryRefSeq) {
+				result = { latestSummaryUpdated: false };
+				return result;
+			}
+
+			// Fetch the latest snapshot and refresh state from it. Note that we need to use the reference sequence number
+			// of the fetched snapshot and not the "summaryRefSeq" that was passed in.
+			const { snapshotTree, snapshotRefSeq: fetchedSnapshotRefSeq } =
+				await fetchLatestSnapshot();
+
+			// Possible re-entrancy. We may have updated latest summary state while fetching the snapshot. If the fetched
+			// snapshot is older than the latest tracked summary, ignore it.
+			if (this.referenceSequenceNumber >= fetchedSnapshotRefSeq) {
+				result = { latestSummaryUpdated: false };
+				return result;
+			}
+
+			await this.refreshLatestSummaryFromSnapshot(
+				fetchedSnapshotRefSeq,
+				snapshotTree,
+				undefined,
+				EscapedPath.create(""),
+				correlatedSummaryLogger,
+				readAndParseBlob,
+			);
+
+			result = {
+				latestSummaryUpdated: true,
+				wasSummaryTracked: false,
+				snapshotTree,
+				summaryRefSeq: fetchedSnapshotRefSeq,
 			};
-			this.defaultLogger.sendTelemetryEvent({
-				eventName: "PendingSummaryNotFound",
+			return result;
+		} finally {
+			const loggingProps: {
+				latestSummaryUpdated?: boolean;
+				wasSummaryTracked?: boolean;
+				summaryRefSeq?: number;
+			} = { ...result };
+			this.logger.sendTelemetryEvent({
```

Comment:
@tyler-cai-microsoft Added the log that will tell the result of refresh latest summary.